src/S3/S3Client.php

```diff
@@ -623,7 +623,9 @@ private function getLocationConstraintMiddleware()
         $region = $this->getRegion();
         return static function (callable $handler) use ($region) {
             return function (Command $command, $request = null) use ($handler, $region) {
-                if ($command->getName() === 'CreateBucket') {
+                if ($command->getName() === 'CreateBucket'
+                    && empty($command['CreateBucketConfiguration']['LocationConstraint'])
+                ) {
                     if ($region === 'us-east-1') {
```

Comment:
How would a client configured to talk to `us-east-1` then specify a location constraint of a different region?